cs_tools/sync/postgres/syncer.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import pydantic
+import sqlalchemy as sa
+import sqlmodel
+from cs_tools.sync.base import DatabaseSyncer
+from typing import TYPE_CHECKING, Any, Literal ,Optional
+from pydantic_core import PydanticCustomError
+import pathlib
+import logging
+from sqlalchemy.orm import sessionmaker
+from cs_tools import __version__
+from cs_tools.sync import utils as sync_utils
+from cs_tools.sync.base import DatabaseSyncer
+
+# if TYPE_CHECKING:
+from cs_tools.sync.types import TableRows
+
+log = logging.getLogger(__name__)
+
+
+class Postgres(DatabaseSyncer):
+    """
+    Interact with Postgres Database
+    """
+    
+    __manifest_path__ = pathlib.Path(__file__).parent / "MANIFEST.json"
+    __syncer_name__ = "postgres"
+   
+    dbname: str 
+    user:  str 
+    password:  str 
+    host:  str 
+    port: int 
+    schema_ : str = pydantic.Field(default="PUBLIC",alias="schema")
+    authentication: Literal["basic", "key-pair", "sso", "oauth"]
+    log_level: Literal["debug", "info", "warning"] = "warning"
+    secret: Optional[str] = None
+    private_key_path: Optional[pydantic.FilePath] = None
+        
+    @pydantic.field_validator("secret", mode="before")
+    def ensure_auth_secret_given(cls, value: Any, info: pydantic.ValidationInfo) -> Any:
+        if info.data.get("authentication") == "sso" or value is not None:
+            return value
+
+        if info.data.get("authentication") == "basic":
+            raise PydanticCustomError("missing", "Field required, you must provide a password", {"secret": value})
+
+        if info.data.get("authentication") == "oauth":
+            raise PydanticCustomError("missing", "Field required, you must provide an oauth token", {"secret": value})
+    
+        
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        logging.getLogger("postgres").setLevel(self.log_level.upper())
+        
+        self._engine = sa.create_engine(self.make_url())
+        self.Session = sessionmaker(bind=self.engine)
+        self.metadata = sqlmodel.MetaData(schema=self.schema_)
+
+    def __repr__(self) -> str:
+        database = self.dbname
+        username = self.user
+        host = self.host
+        port = self.port
+        return f"<PostgresSyncer DATABASE='{database}', USER='{username}', HOST='{host}', PORT='{port}'>"
+    
+    def make_url(self):
+        """Create a connection string for the Postgres JDBC driver."""
+        dialect = "postgresql"
+        driver = "psycopg2"
+        connection_string = f"{dialect}+{driver}://{self.user}:{self.password}@{self.host}/{self.dbname}"
```

Comment:
If the `self.password` attribute is optional, then this will likely fail in some cases.

We can handle it though.

```python
        auth = username if self.password is None else f"{self.username}:{self.password}"
        return f"postgresql+psycopg2://{auth}@{self.host}/{self.dbname}"
```

Additionally, we should make sure `dbname` is switched to `database` to conform to the other database syncers.

Reply:
this block has been replaced with the previous code to have include dynamic approach.